### ISSUE TITLE
Discussion: What happen to a conversation which created by bot and the bot be deleted later on

### ISSUE THREAD

**Issue by namchuai** [janhq/jan#428]
**Motivation**
What will happen to a conversation, which created by a bot and later on, user choose to delete the bot?
In poe.com, they still keep the conversation, and do not allow user to send further message to that conversation.

**Alternatives**
We might have some other UX:
- Remove the conversation which created by that deleted bot (I'm against this)
- Keep the conversation, not allow user to send further message
- Allow user to choose another bot/model to continue with that conversation
_Please add more if you can think of any. I will update and put it here_

**Resources**
![Screenshot 2023-10-23 at 10 17 35](https://github.com/janhq/jan/assets/10397206/64652093-e829-4dfe-9dad-1f25bca3473c)

cc @louis-jan @hiento09 @0xSage @dan-jan

**Comment by 0xSage** [comment]
Thank you for providing the alternative options @nam-john-ho , its super helpful.

Following our current discussions around being OpenAI compatible, it seems we can support (at the API level) `Allow user to choose another bot/model to continue with that conversation`.

Notice that `assistants` are properties of `messages`, rather than of `threads`.
https://platform.openai.com/docs/api-reference/messages/object

Interestingly, the ChatGPT UI doesn't allow switching though.

For now, I vote we keep things simple and go with: `Keep the conversation, not allow user to send further message`. @dan-jan may have different thoughts.

**Comment by 0xSage** [comment]
Changes - based on latest spec: 

1. User start Thread 1 with Assistant A
2. User deletes Assistant A
3. Assistant A icon disappears from Ribbon
4. Historical conversation remains in ./threads


tl;dr: After a thread is initialized, users cannot change models anymore
